AdyenNetworking/APIClient/Request.swift

```diff
@@ -55,15 +54,23 @@ public protocol Request: Encodable {
     /// :nodoc:
     /// The HTTP method.
     var method: HTTPMethod { get }
-    
 }
 
-/// Represents an API response.
+/// Describes a ``Request`` extension to be used for async downloading.
+///
+/// A ``DownloadProgressDelegate`` is provided for progress updates.
+@available(iOS 15.0.0, *)
+public protocol AsyncDownloadRequest: Request {
+    var onProgressUpdate: ((_ progress: Double) -> Void)? { get }
```

Comment:
why not add this closure as an extra parameter to the perform function?

Reply:
Having it on the request allows for the protocol function overload. I think it also nicely separates the callback from the `await` caller syntax of the async function. But I like the suggestion as it would allow for a trailing closure, but then again, it begins to look like completion handler syntax.